cpp/src/Slice/Parser.cpp

```diff
@@ -1330,23 +1336,15 @@ Slice::Container::createInterfaceDef(const string& name, const InterfaceList& ba
     InterfaceDecl::checkBasesAreLegal(name, bases, _unit);
 
     InterfaceDefPtr def = make_shared<InterfaceDef>(shared_from_this(), name, bases);
-    _unit->addContent(def);
```

Comment:
Same for interface. Now we add the `Decl` before we add the `Def`.
This is more expected from the POV of the visitors.